content/src/content/jcr_root/apps/core/wcm/components/carousel/v2/carousel/clientlibs/site/js/carousel.js

```diff
@@ -0,0 +1,641 @@
+
+                if (that._elements["play"]) {
+                    that._elements["play"].addEventListener("click", onPlayClick);
+                }
+
+                window.addEventListener("focus", onFocusChange, true);
+            }
+
+            that._elements.self.addEventListener("keydown", onKeyDown);
+
+            if (!that._properties.autopauseDisabled) {
+                that._elements.self.addEventListener("mouseenter", onMouseEnter);
+                that._elements.self.addEventListener("mouseleave", onMouseLeave);
+            }
+        }
+
+        /**
+         * Handles carousel keydown events
+         *
+         * @private
+         * @param {Object} event The keydown event
+         */
+        function onKeyDown(event) {
+            var index = that._active;
+            var lastIndex = that._elements["indicator"].length - 1;
+
+            switch (event.keyCode) {
+                case keyCodes.ARROW_LEFT:
+                case keyCodes.ARROW_UP:
+                    event.preventDefault();
+                    if (index > 0) {
+                        navigateAndFocusIndicator(index - 1);
+                    }
+                    break;
+                case keyCodes.ARROW_RIGHT:
+                case keyCodes.ARROW_DOWN:
+                    event.preventDefault();
+                    if (index < lastIndex) {
+                        navigateAndFocusIndicator(index + 1);
+                    }
+                    break;
+                case keyCodes.HOME:
+                    event.preventDefault();
+                    navigateAndFocusIndicator(0);
+                    break;
+                case keyCodes.END:
+                    event.preventDefault();
+                    navigateAndFocusIndicator(lastIndex);
+                    break;
+                case keyCodes.SPACE:
+                    if (that._properties.autoplay && (event.target !== that._elements["previous"] && event.target !== that._elements["next"])) {
+                        event.preventDefault();
+                        if (!that._paused) {
+                            pause();
+                        } else {
+                            play();
+                        }
+                    }
+                    if (event.target === that._elements["pause"]) {
+                        that._elements["play"].focus();
+                    }
+                    if (event.target === that._elements["play"]) {
+                        that._elements["pause"].focus();
+                    }
+                    break;
+                default:
+                    return;
+            }
+        }
+
+        /**
+         * Handles focus event on window to check if focus is inside the carousel.
+         * Pauses if focus is within carousel.
+         * Sets {@code Carousel#_hasFocusInside} marker.
+         *
+         * @private
+         * @param {Object} event The focus event
+         */
+
+        function onFocusChange(event) {
```

Comment:
Would make a few minor change here:

• call this `onWindowFocus`.
• make the `_hasFocusInside` variable local to this function rather than adding it to `that` and just call it `hasFocus`.
• not introduce `that._elements["content"]`hook, rather check under the carousel wrapper: `that._elements.self`.